server/server.go

```diff
@@ -865,6 +865,16 @@ func (r *rodeServer) genericList(ctx context.Context, log *zap.Logger, options *
 		return nil, "", createError(log, "error decoding elasticsearch response", err)
 	}
 
+	if options.pageToken != "" || options.pageSize != 0 { // if request is paginated, check for last page
+		_, from, err := esutil.ParsePageToken(nextPageToken)
+		if err != nil {
+			return nil, "", createError(log, "error parsing page token", err)
+		}
+
+		if from >= searchResults.Hits.Total.Value {
```

Comment:
This doesn't seem to work for `ListResources` because it collapses on a field. 

Locally, I have 147 occurrences for only two resources.  If I request without a page size, it does return an empty token; as it gets the default page size of 1000 which is larger than the number of occurrences that are collapsed. But requesting a page size of 2 (all of the resources), still returns a page token.

There's some discussion [here](https://discuss.elastic.co/t/total-hits-with-field-collapsing/105061) about including an aggregate clause in the query to get the true number of collapsed results.

Reply:
I only tested listResources with the two resource that i have and put a page size of 1 and it behaved as i expected. The first page gets the pageToken and the second does not.

I just tested with a page size of 2 and it also behaved as expected. No page token.
![image](https://user-images.githubusercontent.com/43807484/115937927-ba098800-a45e-11eb-9d4e-4370a9c6465c.png)

Reply:
Ohhh i see what's going on. Good catch. Technically we could avoid the collapse query in its entirety...

Reply:
> Technically we could avoid the collapse query in its entirety...

@aalsabag Does this mean we could keep the current logic but change how the query is being done? Just trying to figure out if it is worth my time to figure out how to do what Alex suggested if we think this will be rendered unnecessary in the near future.

Reply:
I'll look into the link above and see what needs to be done. I would rather not get rid of the collapse because it is useful